validator/src/bootstrap.rs

```diff
@@ -421,10 +431,7 @@ pub fn attempt_download_genesis_and_snapshot(
         }
     }
 
-    let (cluster_info, gossip_exit_flag, gossip_service) = gossip.take().unwrap();
-    cluster_info.save_contact_info();
-    gossip_exit_flag.store(true, Ordering::Relaxed);
-    gossip_service.join().unwrap();
```

Comment:
That's right. In this location, gossip is guaranteed to be Some() because of [this](https://github.com/solana-labs/solana/pull/30948/files#diff-e8a460ab9cc255ea2dc59d277ca568a9b94f7a14fa87703979c43943debad885R531) so it looks like someone took the shortcut of just using an unwrap.

In the new version, gossip will also always be Some. In the event someone modifies this code, the old code would have panicked during the unwrap. Now, we will continue on, which I believe is the right behavior in the case where Gossip service was not started. Don't need to shutdown what doesn't exist!

Reply:
I believe it is shutdown before snapshot download. But upon further inspection.. the shutdown at the very end of `rpc_bootstrap` seems unnecessary. I believe Gossip is guaranteed to be `None` at this point. Maybe it was useful in the past when we broke out of the loop after starting Gossip but before attempting download.